### ISSUE TITLE
TSC Meeting (12th Sept 2023)

### ISSUE THREAD

**Issue by nzlosh** [StackStorm/discussions#124]
Sept 2023 @StackStorm/tsc `1 hour` meeting plan:
- See https://github.com/StackStorm/community/issues/33 for the schedule.
- TLDR; Jitsi link: https://meet.jit.si/StackStormTSC

## Attendees

- Carlos
- Chaimaa
- Dale Smith @ Orchestral
- Eugen
- Haven
- Scott

### New Participants

- Scott has recently started using StackStorm for auto-remediation and is interested in St2 progress towards u22.04 py3.10.
- Haven works with Scott and is interested in hardening St2 and running St2 as non-root in containers.
- Chaimaa has worked in the R&D team to deploy St2 to multiple internal and external clients, helped automated daily tasks for the   N.O.C. (network operation centre), has experimented with ChatOps and would like to try NoOps, has expressed interest in contributing to the jira pack and possibly documentation.

### Blog / Web site

Dale presented the proposition to restart the web site migration from WordPress to Eleventy as documented here https://github.com/StackStorm/community/issues/123  with the objective to bring some vitality and better public facing energy to the website.

There are still on-going discussions for and against the use of Eleventy vs Hugo.

### Python 3.10 / Ubuntu 22.04 support

Status quo, progress remains predominantly blocked due to numerous module dependency incompatibilities and would benefit from addition community assistance.  Scott expressed the possibility of getting involved if time permits.

**Comment by AndroxxTraxxon** [comment]
With regards to the Python3.10 dependency incompatibilities, I've got this [version](https://github.com/AndroxxTraxxon/st2/tree/py3_11) running on 3.11. There weren't any major incompatible language or library changes between the two, so dependencies should just work on 3.10

I did run into the mess around gunicorn, but they've finally cut a release after like 3 years, so you should be able to pin the latest tag version instead of the commit that I have there.

I also had to make a [tweak to the Orquesta engine](https://github.com/StackStorm/orquesta/pull/253) to get it working due to the deprecated collections imports being actually removed in Python 3.10

I know none of this is concrete or anything, but StackStorm works in Python 3.10/3.11. We just need to get the test suite updated to actually pass with the current mainline.

**Comment by nzlosh** [comment]
What version of Mongo did you test against?  I was under the impression that there would have to be a lot of code update to bump the mongo dependencies.

**Comment by amanda11** [comment]
> With regards to the Python3.10 dependency incompatibilities, I've got this [version](https://github.com/AndroxxTraxxon/st2/tree/py3_11) running on 3.11. There weren't any major incompatible language or library changes between the two, so dependencies should just work on 3.10
> 
> I did run into the mess around gunicorn, but they've finally cut a release after like 3 years, so you should be able to pin the latest tag version instead of the commit that I have there.
> 
> I also had to make a [tweak to the Orquesta engine](https://github.com/StackStorm/orquesta/pull/253) to get it working due to the deprecated collections imports being actually removed in Python 3.10
> 
> I know none of this is concrete or anything, but StackStorm works in Python 3.10/3.11. We just need to get the test suite updated to actually pass with the current mainline.

That's good to know, that brings it to same state as python 3.9 then. Really pleased that gunicorn release has been done - we've been waiting on that for a while.

**Comment by AndroxxTraxxon** [comment]
> What version of Mongo did you test against?  I was under the impression that there would have to be a lot of code update to bump the mongo dependencies.

I tested against the version included in the ansible-st2 playback, which I think is version 4.x

**Comment by arm4b** [comment]
@AndroxxTraxxon The build in orquesta main branch was fixed and is 🟢 now, so you should be unblocked to work on the https://github.com/StackStorm/orquesta/pull/253. Try to fix the merge conflicts and experiment with the build matrix you're trying to add.

**Comment by AndroxxTraxxon** [comment]
Thanks! I'll give it a go!
